util/cmp/stream.go

```diff
@@ -84,12 +84,12 @@ func WithTarDoneChan(ch chan<- bool) SenderOption {
 	}
 }
 
-// SendRepoStream will compress the files under the given repoPath and send
+// SendRepoStream will compress the files under the given rootPath and send
 // them using the plugin stream sender.
-func SendRepoStream(ctx context.Context, appPath, repoPath string, sender StreamSender, env []string, excludedGlobs []string, opts ...SenderOption) error {
+func SendRepoStream(ctx context.Context, appPath, rootPath string, sender StreamSender, env []string, excludedGlobs []string, opts ...SenderOption) error {
 	opt := newSenderOption(opts...)
 
-	tgz, mr, err := GetCompressedRepoAndMetadata(repoPath, appPath, env, excludedGlobs, opt)
+	tgz, mr, err := GetCompressedRepoAndMetadata(rootPath, appPath, env, excludedGlobs, opt)
```

Comment:
Instead of changing the root path, would it be possible to pass a path into the tgz method to "filter" included files and effectively produce a sparse copy?

It would be nice to preserve the original directory structure of the repo so that the user can write their script to assume working from the repo root instead of a dynamically-calculated root.

Reply:
Note sure about the benefit. Let me elaborate  and please correct me if Im wrong:

`generate & discover`  commands run under application relative path, using `metadata.AppRelPath` to allocate under this directory before run those commands. Then there is no benefit to create the original tgz from the repoPath vs rootPath if the AppRelPath is well calculated no?


Wdyt?

Btw, maybe make sense to update [this doc](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/)


Thanks!